tests/test_encode_decode.py

```diff
@@ -95,8 +97,87 @@ def test_b85_encode_decode_round_trip(self, payload, pad):
 
 
 class TestBinASCII(unittest.TestCase):
-    # TODO: https://docs.python.org/3/library/binascii.html
-    pass
+    @given(payload=st.binary(), backtick=st.booleans())
+    def test_b2a_uu_a2b_uu_round_trip(self, payload, backtick):
+        if sys.version_info[:2] >= (3, 7):
+            x = binascii.b2a_uu(payload, backtick=backtick)
+        else:
+            x = binascii.b2a_uu(payload)
+        self.assertEqual(payload, binascii.a2b_uu(x))
+
+    @given(payload=st.binary(), newline=st.booleans())
+    def test_b2a_base64_a2b_base64_round_trip(self, payload, newline):
+        x = binascii.b2a_base64(payload, newline=newline)
+        self.assertEqual(payload, binascii.a2b_base64(x))
+
+    @given(
+        payload=st.binary(),
+        quotetabs=st.booleans(),
+        istext=st.booleans(),
+        header=st.booleans(),
+    )
+    def test_b2a_qp_a2b_qp_round_trip(self, payload, quotetabs, istext, header):
+        x = binascii.b2a_qp(payload, quotetabs=quotetabs, istext=istext, header=header)
+        self.assertEqual(payload, binascii.a2b_qp(x, header=header))
+
+    @given(payload=st.binary())
+    def test_rlecode_hqx_rledecode_hqx_round_trip(self, payload):
+        x = binascii.rlecode_hqx(payload)
+        self.assertEqual(payload, binascii.rledecode_hqx(x))
+
+    @given(payload=st.binary())
+    def test_b2a_hqx_a2b_hqx_round_trip(self, payload):
+        # assuming len(payload) as 3, since it throws exception: binascii.Incomplete, when length is not a multiple of 3
+        if len(payload) % 3:
+            with self.assertRaises(binascii.Incomplete):
+                x = binascii.b2a_hqx(payload)
+                binascii.a2b_hqx(x)
+            payload += b"\x00" * (-len(payload) % 3)
+        x = binascii.b2a_hqx(payload)
+        res, _ = binascii.a2b_hqx(x)
+        self.assertEqual(payload, res)
+
+    @given(payload=st.binary(), value=st.just(0) | st.integers())
```

Comment:
Final suggestion - let's preserve the pypy issue you found by adding `@example(payload=b"", value=2 ** 63)`.  You'll need to rebase on master for this, `example` is already imported on my branch.

After that I'm happy to merge once PyPy developers have seen this.

Reply:
Sure 👍 

I had a doubt, just to be sure. What exactly do you mean by, rebase on master?

Reply:
https://www.atlassian.com/git/tutorials/rewriting-history and https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase are good introductions :slightly_smiling_face: 

This is intermediate OSS contributor level - I would normally just fix up the history after a new contributor is done, but you've been going really well and so I thought you might be up for learning another new skill!

Reply:
Hey @Zac-HD. Sorry to bother you with something this silly. But, I am getting a lot of conflicts while doing the rebase. Is that normal?

Reply:
I figured it out. 😄

Reply:
Not silly!  It really depends on what you're rebasing - sometimes I have no commits, sometimes so many that I just give up and try something else :sweat_smile: 

Sounds like you worked it out though, I can see you pushed the new version - nice work!